python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -298,22 +298,21 @@ def launch_scheduled_runs(
         elif location_entry.load_error:
             error_locations.add(location_entry.origin.location_name)
 
-    # Remove any schedule states that were previously created with DECLARED_IN_CODE
-    # and can no longer be found in the workspace (so that if they are later added
-    # back again, their timestamps will start at the correct place)
+    # Remove any schedule states that were previously created and can no longer
+    # be found in the workspace (so that if they are later added back again,
+    # their timestamps will start at the correct place)
     states_to_delete = [
         schedule_state
         for selector_id, schedule_state in all_schedule_states.items()
         if selector_id not in schedules
-        and schedule_state.status == InstigatorStatus.DECLARED_IN_CODE
```

Comment:
we discussed a bit in slack but the other thing i was wondering is whether we should only do this if, say, >12 hrs has passed since the stored last_iteration_timestamp or something, just to guard against somebody wiping out all their schedules by accidentally pushing a change that removes them but isn't considered an error (we would need to make sure we are always updating the last_iteration_timestamp, even when the schedule is stopped - which we probably want to do anyway for the start-stop-start again case)

Reply:
We wouldn't want to land that unless we did the full iteration timestamp update across all states?

Reply:
yeah, something like this? https://github.com/dagster-io/dagster/pull/24603 doesn't seem too bad

Reply:
I guess the downside of not doing the 12 hour period is that the schedule doesn't tick for the time period that it was removed from the workspace?  Or that it's status is different than expected?  That doesn't seem that bad to me, honestly.

Reply:
I would defer to you though... were we just being conservative when we restricted this logic to only apply to declared in code status?